test/e2e/storage/drivers/in_tree.go

```diff
@@ -125,7 +125,9 @@ func (n *nfsDriver) SkipUnsupportedTest(pattern storageframework.TestPattern) {
 
 func (n *nfsDriver) GetVolumeSource(readOnly bool, fsType string, e2evolume storageframework.TestVolume) *v1.VolumeSource {
 	nv, ok := e2evolume.(*nfsVolume)
-	framework.ExpectEqual(ok, true, "Failed to cast test volume to NFS test volume")
+	if !ok {
+		framework.Failf("Failed to cast test volume of size %v to the NFS test volume", nv)
```

Comment:
Printing `nv` here is not useful, it will always be nil when `ok` is `false`.

Instead include information that shows why the cast failed:
```suggestion
		framework.Failf("Failed to cast test volume of type %T to the NFS test volume", e2evolume)
```

The same applies below.

Reply:
I have updated all similar cases. please have a look.